apis/core.oam.dev/v1alpha1/garbagecollect_types.go

```diff
@@ -66,6 +67,29 @@ type ResourcePolicyRuleSelector struct {
 	OAMResourceTypes []string `json:"oamTypes"`
 	TraitTypes       []string `json:"traitTypes"`
 	ResourceTypes    []string `json:"resourceTypes"`
+	ResourceNames    []string `json:"resourceNames"`
+}
+
+// Match check if current rule selector match the target resource
+func (in ResourcePolicyRuleSelector) Match(manifest *unstructured.Unstructured) bool {
```

Comment:
```suggestion
func (in *ResourcePolicyRuleSelector) Match(manifest *unstructured.Unstructured) bool {
```

It's save for a object method to use pointer to avoid "value copy issues".

Reply:
https://stackoverflow.com/questions/25382073/defining-golang-struct-function-using-pointer-or-not

Reply:
OK.